storage/internal/tempdir/tempdir.go

```diff
@@ -148,7 +149,7 @@ func RecoverStaleDirs(rootDir string) error {
 			continue
 		}
 
-		if rmErr := os.RemoveAll(tempDirPath); rmErr != nil {
+		if rmErr := system.EnsureRemoveAll(tempDirPath); rmErr != nil && !os.IsNotExist(rmErr) {
```

Comment:
As the function doc says:

> This should not return a `os.ErrNotExist` kind of error under any circumstances

So there should be no need to catch this error and if there is do not use os.IsNotExist(), as that funtion says:
> New code should use errors.Is(err, fs.ErrNotExist).